### PR TITLE
Updated form inputs to atomic design principles

### PR DESCRIPTION
**Pull request by jimmynotjim**
Updates form inputs as well as a few necessary parent molecules and organisms of the inputs.

## Additions

- Added `text-input` atom to combine all text style inputs as one and reduce specificity of existing inputs.
- Added `checkbox` and `radio` atoms to simplify styles. Removes the need for hiding the input and relying on pseudo elems on the label.
- Added `label` atom with heading modifier.

## Removals

- Removed form variables from cf-core

## Changes

- Updated form variable naming scheme to be more consistent with each other and our BEM style principles.

## Testing

- run `npm run cf-link`
- in a tab with the sandbox repo run the following:
    - checkout the `form-examples` [branch](https://github.com/cfpb/capital-framework-sandbox/pull/23) 
    - run `npm run cf-link`
    - run `npm run build`
    - run `npm start`
    - navigate to http://localhost:3000/components/cf-forms/ and check out the text inputs at the top and the Atomic Filter Form at the bottom. Ignore all the broken bits, those are fixed and ready for a follow up PR.

## Review

- @mistergone 
- @cfarm 
- @KimberlyMunoz 
- @Scotchester 

## Screenshots

__Labels__

screen shot 2016-08-30 at 4 01 10 pm

__Basic inputs__

screen shot 2016-08-30 at 4 01 30 pm

__Input states__

screen shot 2016-08-30 at 4 01 52 pm

__Filter form example__

screen shot 2016-08-30 at 4 02 22 pm

## Notes

- ⚠️  __WARNING:__ The sandbox branch breaks a lot of the other form elements. There are upcoming PRs to fix those, please ignore for the time being.

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/cf-forms/src/atoms/label.less [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.a-label {
+    display: inline-block;
+}
+
+.a-label__heading {
+    .heading-5();
+
+    // Overwrites heading-5 margin
```
Comment (jimmynotjim) [inline src/cf-forms/src/atoms/label.less]: This makes me question if margin should be set in `heading-5()` vs `h5()`

File: src/cf-forms/src/atoms/text-input.less [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        outline: 1px solid @input-border__warning;
+    }
+
+    &__success {
+        border-color: @input-border__success;
+        outline: 1px solid @input-border__success;
+    }
+}
+
+// Overrides extra left padding.
```
Comment (jimmynotjim) [inline src/cf-forms/src/atoms/text-input.less]: The following were already set up like this and I wasn't sure if they should be chained to text inputs specifically or left open like this. I'm open to suggestions.

File: src/cf-forms/src/cf-forms.less [hunk 3]
```diff
@@ -212,16 +112,16 @@ textarea {
     font-size: unit(@font-size / @base-font-size-px, em );
 }
 
-.error + .@{cf-forms_input-icon-class} {
-    color: @input-error;
+.a-text-input__error + .@{cf-forms_input-icon-class} {
```
Comment (jimmynotjim) [inline src/cf-forms/src/cf-forms.less]: Icons felt out of scope, so they're left here but the correct input classes have been updated.

File: src/cf-forms/src/organisms/form.less [hunk 4]
```diff
@@ -0,0 +1,16 @@
+.o-form {
+    &_group {
+        margin-bottom: 30px;
+    }
+
+    &_fieldset {
+        // Overwrite Normalize
```
Comment (jimmynotjim) [inline src/cf-forms/src/organisms/form.less]: Should we create our own normalize that just has what we want/need? There's also a style on `type=search` inputs that sets the `box-sizing` to `content-box` which has to be overridden with the type attr to get past the specificity.

**Comment by jimmynotjim** [comment]
I think this is ready for review now. Be sure to pull the latest.

File: src/cf-forms/src/organisms/form.less (hunk 4 above)
Comment (mistergone) [inline src/cf-forms/src/organisms/form.less]: That doesn't sound like a bad idea to me. (But out of scope for this PR.)

**Comment by mistergone** [comment]
This looks good to me. 👍 

Note, however, I don't feel like I have good insight to have useful opinions about margins in heading-5 or the placeholder modifiers, for what it's worth. 😁

File: src/cf-forms/src/organisms/form.less (hunk 4 above)
Comment (cfarm) [inline src/cf-forms/src/organisms/form.less]: yeah let's customize our normalize

**Comment by cfarm** [comment]
@mistergone make sure to push that lil merge button after you thumbs up a PR. for now, I get the pleasure of doing it on this one,mwaha

**Pull request merged**